com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Camera/HDCameraUI.Drawers.cs

```diff
@@ -76,6 +79,10 @@ enum ShutterSpeedUnit
             new Vector2(70.41f, 52.63f)
         };
 
+        // Saves the value of the sensor size when the user switches from "custom" size to a preset per camera.
+        // We use a ConditionalWeakTable instead of a Dictionary to avoid keeping alive (with strong references) deleted cameras
+        static ConditionalWeakTable<Camera, object> s_PerCameraSensorSizeHistory = new ConditionalWeakTable<Camera, object>();
```

Comment:
Nice! 

Does it need to be an object and not a vector2 though? I guess it is for checking if the value is there or not? If that's the case TryGetValue returns false if the object is not found.

Reply:
Also (def not needed, but for future reference) to slightly decrease the cost of the hashing and memory consumption of this table, you could use the instance id of the object (Camera) as a Key which is unique per unity object. (Can query from the game object with GetInstanceID() )

Reply:
Unfortunately ConditionalWeakTable work only with reference types, so I could not use Vector2 directly, it does not even compile. There is some overhead with the boxing operation, but this is paid only when the user switches from a preset to custom in the UI. Scripts that add/delete cameras will never hit that.

Using a traditional dictionary + instanceID is certainly an alternative, not sure if it is better or worse.